### PR TITLE
Android deferred deeplinks integration

### PR DESCRIPTION
**Pull request by pzduniak**
cc @keybase/y2ksquad

### CODE REVIEW

File: go/service/deferred_deep_links.go [hunk 1]
```diff
@@ -39,48 +36,21 @@ func (c installReferrerHandler) CallbackWithString(s string) {
...
-		m.Warning("Error in parseInstallReferrer: %v", err)
-		return
-	}
-	if !found {
-		m.Debug("installReferrerHandler#CallbackWithString: invite not found")
-		return
-	}
+	// Play Store seems to replace our URL-encoded plus sign with a space, so we
+	// need to work around it.
+	s = strings.Replace(s, " ", "+", -1)
```
Comment (AMarcedone) [inline go/service/deferred_deep_links.go:41]: It seems + can also be used to encode spaces, which is why the play store is decoding it as a space. I think the fix here is to change the way the website encodes this. Can you check if this is possible? 
More info here: https://developers.google.com/maps/documentation/urls/url-encoding

File: shared/actions/deeplinks.tsx [hunk 2]
```diff
@@ -135,6 +136,16 @@ const handleKeybaseLink = (action: DeeplinksGen.HandleKeybaseLinkPayload) => {
   ]
 }
 
+const handleServiceAppLink = (action: EngineGen.Keybase1NotifyServiceHandleKeybaseLinkPayload) => {
+  const link = action.payload.params.link
+  if (action.payload.params.deferred && !link.startsWith('keybase://team-invite-link/')) {
```
Comment (AMarcedone) [inline shared/actions/deeplinks.tsx:141]: the referrer link coming from the play store is potentially malicious. Whoever reviews the tsx part should make sure that strings which start with `keybase://team-invite-link/` cannot have unintended consequences during their handling.

File: go/service/deferred_deep_links.go (hunk 1 above)
Comment (pzduniak) [reply go/service/deferred_deep_links.go:41]: >For example, 上海+中國 in UTF-8 would be URL-encoded as %E4%B8%8A%E6%B5%B7%2B%E4%B8%AD%E5%9C%8B
`https://play.google.com/store/apps/details?id=io.keybase.ossifrage&referrer=keybase%3A%2F%2Fteam-invite-link%2F5b81a5512e0ef2e9c4afe71921c2e727%2Fe6afb2k%2Bgx2ypqqv9mw9kpebhhhy`
If we run into any issues with this code, we'll just run it only on crypto invite links.

Comment (AMarcedone) [reply go/service/deferred_deep_links.go:41]: Maybe we are decoding twice somehow? 
I agree the scope of this code is limited, but it seems weird that the playstore has such a weird decoding bug.

File: shared/actions/deeplinks.tsx (hunk 2 above)
Comment (pzduniak) [reply shared/actions/deeplinks.tsx:141]: To address this: `actions/deeplinks.tsx` will be changed if I get to finish this, as of right now there is no code for handling deeplinks if you're logged out.

**Pull request merged**